compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/methodgen/MethodGen.java

```diff
@@ -754,6 +754,9 @@ private void generateFrameClassFieldLoad(List<BIRVariableDcl> localVars, MethodV
                 mv.visitFieldInsn(GETFIELD, frameName, localVar.jvmVarName,
                         GET_BSTRING);
                 mv.visitVarInsn(ASTORE, index);
+            } else if (bType.tag == TypeTags.NEVER) {
+                mv.visitFieldInsn(GETFIELD, frameName, localVar.jvmVarName, GET_OBJECT);
+                mv.visitVarInsn(ASTORE, index);
```

Comment:
Shall we move this into `generateFrameClassFieldLoadByTypeTag` method?

Reply:
We already had such an implementation. The issue happens when we call `isAssignable()` on the XML type tag which returns true to the `never` type. Therefore I moved the `never` check before the XML check.